mobile/examples/whisper/android/azure/app/src/main/java/ai/onnxruntime/example/speechrecognition/AudioTensorSource.kt

```diff
@@ -0,0 +1,103 @@
+package ai.onnxruntime.example.speechrecognition
+
+import ai.onnxruntime.OnnxJavaType
+import ai.onnxruntime.OnnxTensor
+import ai.onnxruntime.OrtEnvironment
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import android.util.Log
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.util.concurrent.atomic.AtomicBoolean
+
+class AudioTensorSource {
+    companion object {
+        private const val sampleRate = 16000
+        private const val maxAudioLengthInSeconds = 30
+
+        fun fromRawWavBytes(rawBytes: ByteArray): OnnxTensor {
+            val rawByteBuffer = ByteBuffer.wrap(rawBytes)
+            // TODO handle big-endian native order...
+            if (ByteOrder.nativeOrder() != ByteOrder.LITTLE_ENDIAN) {
+                throw NotImplementedError("Reading Wav data is only supported when native byte order is little-endian.")
+            }
+            rawByteBuffer.order(ByteOrder.nativeOrder())
+            val numBytes = minOf(rawByteBuffer.capacity(), maxAudioLengthInSeconds * sampleRate)
+            val env = OrtEnvironment.getEnvironment()
+            return OnnxTensor.createTensor(
+                env, rawByteBuffer, tensorShape(numBytes.toLong()), OnnxJavaType.UINT8)
+        }
+
+        @SuppressLint("MissingPermission")
+        fun fromRecording(stopRecordingFlag: AtomicBoolean): OnnxTensor {
+            val recordingChunkLengthInSeconds = 1
+
+            val minBufferSize = maxOf(
+                AudioRecord.getMinBufferSize(
+                    sampleRate,
+                    AudioFormat.CHANNEL_IN_MONO,
+                    AudioFormat.ENCODING_PCM_8BIT,
+                ),
+                2 * recordingChunkLengthInSeconds * sampleRate * 1
```

Comment:
nit: the `* 1` doesn't seem to add much meaning. maybe remove it or use a constant named something like `bytesPerSample`.